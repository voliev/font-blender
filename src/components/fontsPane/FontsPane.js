import React from "react";
import TitleFilter from "./TitleFilter";
import SortingFilter from "./SortingFilter";
import CategoriesFilter from "./CategoriesFilter";
import FontsList from "./FontsList";
import SelectedFontsContainer from "./SelectedFontsContainer";
import "./styles/FontsPane.css";

const FontsPane = () => (
  <div className="FontsPane">
    <TitleFilter />
    <CategoriesFilter />
    <SortingFilter />
    <FontsList />
    <SelectedFontsContainer />
  </div>
);

export default FontsPane;
