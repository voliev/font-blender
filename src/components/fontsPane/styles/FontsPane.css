.FontsPane {
  background-color: var(--off-white);
  display: grid;
  grid-template-rows: 4rem 4rem calc(100vh - 15rem) 3rem;
  grid-template-columns: 18rem 4rem;
  grid-template-areas:
    "search icon"
    "sorting sorting"
    "list list"
    "selected selected";
  height: calc(100vh - 4rem);
  position: relative;
  width: var(--side-pane-width);
}
