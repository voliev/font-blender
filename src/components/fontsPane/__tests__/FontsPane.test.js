import React from "react";
import { shallow } from "enzyme";
import FontsPane from "../FontsPane";

it("should render without crashing", () => {
  const wrapper = shallow(<FontsPane />);
  expect(wrapper).toMatchSnapshot();
});

it("should render children components", () => {
  const wrapper = shallow(<FontsPane />);
  expect(wrapper.children().length).toBe(5);
});
