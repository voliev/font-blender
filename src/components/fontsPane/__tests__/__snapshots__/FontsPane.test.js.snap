// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div
  className="FontsPane"
>
  <Connect(TitleFilter) />
  <CategoriesFilter />
  <Connect(SortingFilter) />
  <Connect(FontsList) />
  <Connect(SelectedFonts) />
</div>
`;
